### ISSUE TITLE
Add Resampling Option of Time Stamps

### ISSUE THREAD

**Issue by shanlyMIDE** [MideTechnology/endaq-python-ide#27]
Add a parameter to the `to_pandas` function to allow the user to manipulate the timestamps:

- 'unmodified' (Default) - current behavior
- 'constant' - assume a fixed sample rate and recompute them with `linspace`
- 'resample' - use scipy.signal.resample to resample 

@CFlaniganMide @SamRagusa  @CrepeGoat  @StokesMIDE

**Comment by CrepeGoat** [comment]
relevant meeting notes: https://midetech.atlassian.net/wiki/spaces/EC/pages/1263894529/2021-05-18+Open+Source+Meeting+Notes

**Comment by shanlyMIDE** [comment]
As I'm prepping for the webinar I wanted to bubble this into a conversation again. I noticed this in the plotting functions @SamRagusa has added that there are occasional gaps:
![newplot (77)](https://user-images.githubusercontent.com/35080650/139947116-0ee4db61-a39e-4dad-ab08-b38da78df263.png)

This seems to be because of our non-uniform sampling. If I resample the data uniformly before plotting, those gaps go away:
![newplot (78)](https://user-images.githubusercontent.com/35080650/139947177-26bb38c5-d5e9-40c3-ae29-a38f886cf20b.png)

I could see some hesitation to have the full-on `resample` option using scipy because it uses scipy (although I have opinions on that matter). But at least allowing the option to get uniform spacing with numpy would be nice. 

I've had to add a FAQ in our help center on this issue because it continues to create confusion (and sometimes borderline "alarm") by certain users: https://support.endaq.com/article/125-troubleshooting#general

**Comment by SamRagusa** [comment]
I agree this would be a very useful thing to have.   I've had numerous issues related to these non-uniform time steps, and it's often not obvious that they're the cause (for example, what Steve just discovered).    

Given that it'd be completely optional and opt-in rather as a default, I can't think of a reason it wouldn't be an easy win.

I'm all for it.
